src/App.test.js

```diff
@@ -1,8 +1,82 @@
-import { render, screen } from '@testing-library/react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom'
+import { getAllMovieData } from './apiCalls.js'
+import { getSingleMovieData } from './apiCalls.js'
 import App from './App';
+jest.mock('./apiCalls.js')
 
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
-});
+describe('App', () => {
+  
+  beforeEach(() => {
+    getAllMovieData.mockResolvedValueOnce({ 'movies': [{
+      "id": 694919,
+      "poster_path": "https://image.tmdb.org/t/p/original//6CoRTJTmijhBLJTUNoVSUNxZMEI.jpg",
+      "backdrop_path": "https://image.tmdb.org/t/p/original//pq0JSpwyT2URytdFG0euztQPAyR.jpg",
+      "title": "Money Plane",
+      "average_rating": 6.666666666666667,
+      "release_date": "2020-09-29"
+      },
+      {
+      "id": 337401,
+      "poster_path": "https://image.tmdb.org/t/p/original//aKx1ARwG55zZ0GpRvU2WrGrCG9o.jpg",
+      "backdrop_path": "https://image.tmdb.org/t/p/original//zzWGRw277MNoCs3zhyG3YmYQsXv.jpg",
+      "title": "Mulan",
+      "average_rating": 4.909090909090909,
+      "release_date": "2020-09-04"
+      }]
+    })
+  })
+
+  it('should render a navbar, show loading movies text', () => {
+    render(<App />);
+    const navbarElement = screen.getByTestId('navbar-element');
+    expect(navbarElement).toBeInTheDocument();
+
+    const loadingMessage = screen.getByText('...Loading Movies...');
+    expect(loadingMessage).toBeInTheDocument();
+  })
+
+  it('should render all movies', async () => {
+    render(<App />);
+    const moviesElement = screen.getByTestId('movies-element');
+    expect(moviesElement).toBeInTheDocument();
+
+    const newMovie = await waitFor(() => {
+      screen.getByText('Money Plane')
+      screen.getByText('Mulan')
+    })
+  })
+
+  it('should allow users to click on a movie and take them to a MOvie Preview', async () => {
```

Comment:
Test is well written and clear what our assertions, executions, and expectations are.